### ISSUE TITLE
Ability to make fs-backup without demonset

### ISSUE THREAD

**Issue by tropnikovvl** [vmware-tanzu/velero#8031]
**Describe the problem/challenge you have**

I use Velero in a shared Kubernetes cluster, but I am faced with the fact that daemonsets are located on all nodes (~100) when the application has only 2-3 replicas.
Thus, I take away computer resources from other applications and my security team is unhappy that Velero requires [broad access rights to directories on nodes](https://velero.io/docs/v1.14/file-system-backup/#limitations), which can create potential vulnerabilities and gain unauthorized access to other application's data.

**Describe the solution you'd like**
So I wanted to know if it is possible to make an fs-backup without a daemonset? For example, some kind of job or pod?

**Environment:**

- Velero version (use `velero version`): v1.14.0
- Kubernetes version (use `kubectl version`): v1.29.4
- Kubernetes installer & version:
- Cloud provider or hardware configuration: AWS EKS
- OS (e.g. from `/etc/os-release`): Amazon Linux 2023

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by Lyndon-Li** [comment]
The problems you described are not about the deamonset itself.

> I take away computer resources from other applications
The deamonset indeed runs in every node, but if there is no backup/restore in a node, the daemonset in that node takes very little resource. For fs-backup, the backup/restore only happens in the nodes where the workload pods run.

> my security team is unhappy that Velero requires broad access rights to directories on nodes
This is a known limitation for fs-backup. And this also happens for data mover backup/restore at present. In 1.15, this design #7576 will fix the problem for data mover backup/restore.
Therefore, the recommendation here is that whenever data mover backup/restore is possible, use it over fs-backup, because fs-backup is not a consistent backup.
From 1.15, the limitation you concerned will be fixed for data mover backup/restore.

**Comment by tropnikovvl** [comment]
@Lyndon-Li 
Thanks for the detailed answer!

Can you tell me why the fs-backup is inconsistent? I have not seen such information.
And approximate timing (if you have such information) for introducing new functionality with the date mover?

**Comment by Lyndon-Li** [comment]
> Can you tell me why the fs-backup is inconsistent?

fs-backup reads data from the live file system, so the data is not consistent. See [here](https://velero.io/docs/v1.14/csi-snapshot-data-movement/#:~:text=Besides%2C%20Velero%20File%20System%20Backup%20which%20could%20also%20back%20up%20the%20volume%20data%20to%20a%20pre%2Ddefined)

**Comment by Lyndon-Li** [comment]
> And approximate timing (if you have such information) for introducing new functionality with the date mover?

CSI snapshot data mover backup is already there since 1.12

**Comment by tropnikovvl** [comment]
@Lyndon-Li 
Do I understand correctly that you recommend using a `data mover` instead of an `fs-backup` because it takes an instant and consistent snapshot of the disk at a point in time?

The documentation also says that now the date mover only works with the daemonset, you mentioned version 1.15 and apparently this limitation will be removed in it.

Correct me if I am wrong.

**Comment by Lyndon-Li** [comment]
> The documentation also says that now the date mover only works with the [daemonset](https://velero.io/docs/v1.14/csi-snapshot-data-movement/#install-velero-node-agent), you mentioned version 1.15 and apparently this limitation will be removed in it.

The data mover and fs-backup will always work only with the daemonset. But in 1.15 your two concerns as you mentioned will be removed for data mover backup.

**Comment by tropnikovvl** [comment]
@Lyndon-Li 
I understand this may be difficult, but can you predict the approximate release date for 1.15?
I don't need exact and official data, just guesses so I can plan my work

**Comment by Lyndon-Li** [comment]
See the [roadmap of 1.15](https://github.com/vmware-tanzu/velero/wiki/1.15-Roadmap)

**Comment by tropnikovvl** [comment]
@Lyndon-Li 
I read the [document](https://github.com/vmware-tanzu/velero/blob/main/design/vgdp-micro-service/vgdp-micro-service.md#node-agent) and wanted to ask one last clarifying question.

If I plan to use only CSI snapshots, does this mean that I don't have to deploy the node-agent daemonset?
If I understand correctly, there will only be a controller for `fs-backup` and not for snapshots.

**Comment by Lyndon-Li** [comment]
Yes, if you use only CSI snapshots, you don't need to install node-agent. But CSI snapshots means storage the snapshot dat a locally in the same storage location as the source volume, no data is moved to the BSL

**Comment by tropnikovvl** [comment]
@Lyndon-Li 
You mean that the backup metadata will be stored in s3 (as an example), and the data itself will be stored in the same storage (VolumeSnapshotClass) that the PVC provides, only now it will be a snapshot. Right?
https://velero.io/docs/main/csi/

**Comment by Lyndon-Li** [comment]
> @Lyndon-Li You mean that the backup metadata will be stored in s3 (as an example), and the data itself will be stored in the same storage (VolumeSnapshotClass) that the PVC provides, only now it will be a snapshot. Right? https://velero.io/docs/main/csi/

Yes
